Bercea_Bogdan/Week_2/Domain/Domain/CreateEmployeeOp/CreateEmployeeResult.cs

```diff
@@ -0,0 +1,36 @@
+﻿using CSharp.Choices.Attributes;
+using Domain.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Domain.Domain.CreateEmployeeOp;
+
+namespace Domain.Domain.CreateEmployeeOp
+{
+    [AsChoice]
+    public static partial class CreateEmployeeResult
+    {
+        public interface ICreateEmployeeResult { }
+
+        public class EmployeeCreated : ICreateEmployeeResult
+        {
+            public Employee Employee { get; }
+
+            public EmployeeCreated(Employee employee)
+            {
+                Employee = employee;
+            }
+        }
+
+        public class EmployeeNotCreated : ICreateEmployeeResult
+        {
```

Comment:
The sum types don't need to always segregate to Created/NotCreated. When you model your domain, try to find the actual cases that apply to your design.